factoryWithTypeCheckers.js

```diff
@@ -434,6 +441,9 @@ module.exports = function(isValidElement, throwOnDirectAccess) {
       var allKeys = assign({}, props[propName], shapeTypes);
       for (var key in allKeys) {
         var checker = shapeTypes[key];
+        if (shapeTypes.hasOwnProperty(key) && typeof checker !== 'function') {
+          return invalidValidatorError(componentName, location, propFullName, key, typeof checker);
```

Comment:
instead of using `typeof` here, can we use `getPreciseType`?
```suggestion
          return invalidValidatorError(componentName, location, propFullName, key, getPreciseType(checker));
```

Reply:
Yep!